plugin/src/main/java/org/gradle/testretry/internal/executer/framework/TestFrameworkStrategy.java

```diff
@@ -25,25 +26,48 @@
 import org.gradle.util.GradleVersion;
 
 import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Iterator;
+import java.util.regex.Pattern;
 
 /**
  * Instances are scoped to a test task execution and are reused between rounds.
  */
 public interface TestFrameworkStrategy {
 
+    Pattern SPOCK2_CORE_JAR_NAME_PATTERN = Pattern.compile(".*/spock-core-2.*\\.jar");
+
     @Nullable
-    static TestFrameworkStrategy of(TestFramework testFramework) {
+    static TestFrameworkStrategy of(JvmTestExecutionSpec spec) {
+        TestFramework testFramework = spec.getTestFramework();
         if (testFramework instanceof JUnitTestFramework) {
             return new JunitTestFrameworkStrategy();
         } else if (testFramework instanceof JUnitPlatformTestFramework) {
-            return new Junit5TestFrameworkStrategy();
+            return new Junit5TestFrameworkStrategy(isSpock2Used(spec));
         } else if (testFramework instanceof TestNGTestFramework) {
             return new TestNgTestFrameworkStrategy();
         } else {
             return null;
         }
     }
 
+    static boolean isSpock2Used(JvmTestExecutionSpec spec) {
+        Iterator<? extends File> classpathIterator = spec.getClasspath().iterator();
```

Comment:
❓ Don't we also have to check `spec.getModulePath()` since Spock 2 should end up there if `java.modularity.inferModulePath` is set to `true` in the build script?

Reply:
It seems like setting it to `true` is not enough. Does it have to be a similar setup as described in https://docs.gradle.org/current/samples/sample_java_modules_multi_project_with_integration_tests.html? Where the test source set is a module itself?

Reply:
I added a check of module path in https://github.com/gradle/test-retry-gradle-plugin/pull/202/commits/d06b3752e9240e2f2497ca6fbd36e9b6ae50f674. But I have not managed to set up a test yet where that would happen. I can look into it once I am back.